src/io/binary_parser.cpp

```diff
@@ -156,32 +156,84 @@ long long read_variables(string filename, dd::FactorGraph &fg)
     return count;
 }
 
-long long read_factors(string filename, dd::FactorGraph &fg)
+long long read_factors(string filename, dd::FactorGraph &fg, int inc)
 {
-    ifstream file;
-    file.open(filename.c_str(), ios::in | ios::binary);
-    long long count = 0;
-    long long id;
-    long long weightid;
-    short type;
-    long long edge_count;
-    while (file.good()) {
-        file.read((char *)&id, 8);
-        file.read((char *)&weightid, 8);
-        file.read((char *)&type, 2);
-        if (!file.read((char *)&edge_count, 8)) break;
+    if (inc) {
```

Comment:
I think this flag is a temporary solution. Two functions could be a good way to make it clear.